bindings/rust/s2n-tls/src/config.rs

```diff
@@ -488,6 +489,46 @@ impl Builder {
         Ok(self)
     }
 
+    /// Sets a custom callback which provides access to session tickets when they arrive
+    pub fn set_session_ticket_callback<T: 'static + SessionTicketCallback>(
+        &mut self,
+        handler: T,
+    ) -> Result<&mut Self, Error> {
+        // Define C callback function that can be set on the s2n_config struct
+        unsafe extern "C" fn session_ticket_cb(
+            conn_ptr: *mut s2n_connection,
+            _context: *mut ::libc::c_void,
+            session_ticket: *mut s2n_session_ticket,
+        ) -> libc::c_int {
+            let session_ticket = NonNull::new(session_ticket).expect("ticket should not be null");
+            with_context(conn_ptr, |conn, context| {
+                let callback = context.session_ticket_callback.as_ref();
+                callback.map(|c| {
```

Comment:
If we're willing to give up a little bit of client flexibility, could we actually allocate the memory for the session ticket here and truly have it be an owned object? That way the client can just directly put it in a map or whatever, rather than having to manually copy it each time.